atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/command/CleanTransactionRange.java

```diff
@@ -70,50 +60,41 @@ public int execute(AtlasDbServices services) {
-
+        Multimap<Cell, Long> toDelete = HashMultimap.create();
         long maxTimestamp = startTimestampExclusive;
         while (range.hasNext()) {
             RowResult<Value> row = range.next();
-            long startResult = TransactionConstants.getTimestampForValue(row.getRowName());
+            byte[] rowName = row.getRowName();
+            long startResult = TransactionConstants.getTimestampForValue(rowName);
             maxTimestamp = Math.max(maxTimestamp, startResult);
-            long endResult = 0L;
-            if (row.getOnlyColumnValue() != null) {
-                endResult = TransactionConstants.getTimestampForValue(row.getOnlyColumnValue().getContents());
-                maxTimestamp = Math.max(maxTimestamp, endResult);
-            } else {
+            
+            Value value;
+            try {
+                value = row.getOnlyColumnValue();
+            } catch (IllegalStateException e){
                 //this should never happen
-                System.out.printf("Error: Found a row in the transactions table that didn't have 1 and only 1 column value: start=%d", startResult);
+                System.out.printf("Error: Found a row in the transactions table that didn't have 1 and only 1 column value: start=%d\n", startResult);
+                continue;
             }
 
-            System.out.printf("Found and cleaning possibly inconsistent transaction: [start=%d, commit=%d]", startResult, endResult);
+            long endResult = TransactionConstants.getTimestampForValue(value.getContents());
+            maxTimestamp = Math.max(maxTimestamp, endResult);
+            System.out.printf("Found and cleaning possibly inconsistent transaction: [start=%d, commit=%d]\n", startResult, endResult);
 
-            for (Cell cell : row.getCellSet()) {
-                toClean.put(cell, cleaned);
-            }
+            Cell key = Cell.create(rowName, TransactionConstants.COMMIT_TS_COLUMN);
+            toDelete.put(key, value.getTimestamp());  //value.getTimestamp() should always be 0L but this is safer
         }
 
-        if (!toClean.isEmpty()) {
-            if (delete) {
-                Multimap<Cell, byte[]> toCleanMulti = Multimaps.forMap(toClean);
-                Multimap<Cell, Long> toDelete = Multimaps.transformValues(toCleanMulti, new Function<byte[], Long>() {
-                    public Long apply(byte[] in) {
-                        return TransactionConstants.getTimestampForValue(in);
-                    }
-                });
-                kvs.delete(TransactionConstants.TRANSACTION_TABLE, toDelete);
-                System.out.println("\nDelete completed.");
-            } else {
-                kvs.putUnlessExists(TransactionConstants.TRANSACTION_TABLE, toClean);
-                System.out.println("\nClean completed.");
-            }
+        if (!toDelete.isEmpty()) {
+            kvs.delete(TransactionConstants.TRANSACTION_TABLE, toDelete);
```

Comment:
Because we think the size of the deletes can potentially to be too large to fit in memory?  In that case we should actually be batching the original getRange call.  Either way I can do that in a follow-up CR.